draft-ietf-add-ddr.md

```diff
@@ -171,44 +177,50 @@ Encrypted Resolver name.
 
 ## Authenticated Discovery {#authenticated}
 
+If the Unencrypted Resolver's IP address is a Public IP, it is considered a
+global reference identity and is subject to cryptographic authentication.
+
 In order to be considered an authenticated Designated Resolver, the
-TLS certificate presented by the Encrypted Resolver MUST contain both the domain
-name (from the SVCB answer) and the IP address of the designating Unencrypted
+TLS certificate presented by the Encrypted Resolver MUST contain the IP address
+of the designating Unencrypted
 Resolver within the SubjectAlternativeName certificate field. The client MUST
-check the SubjectAlternativeName field for both the Unencrypted Resolver's IP
-address and the advertised name of the Designated Resolver. If the
+check the SubjectAlternativeName field for the Unencrypted Resolver's IP
+address. If the
 certificate can be validated, the client SHOULD use the discovered Designated
 Resolver for any cases in which it would have otherwise used the
-Unencrypted Resolver. If the Designated Resolver has a different IP
-address than the Unencrypted Resolver and the TLS certificate does not cover the
-Unencrypted Resolver address, the client MUST NOT use the discovered Encrypted
+Unencrypted Resolver. Otherwise, the client SHOULD NOT use the discovered Encrypted
 Resolver. Additionally, the client SHOULD suppress any further queries for
 Designated Resolvers using this Unencrypted Resolver for the length of
 time indicated by the SVCB record's Time to Live (TTL).
 
-If the Designated Resolver and the Unencrypted Resolver share an IP
-address, clients MAY choose to opportunistically use the Encrypted Resolver even
-without this certificate check ({{opportunistic}}).
-
 ## Opportunistic Discovery {#opportunistic}
 
-There are situations where authenticated discovery of encrypted DNS
-configuration over unencrypted DNS is not possible. This includes Unencrypted
-Resolvers on non-public IP addresses whose identity cannot be confirmed using
-TLS certificates.
+If the Unencrypted Resolver's IP address is a Private IP, it is considered
+a local reference identity that cannot be confirmed using TLS certificates.
+Instead, the ability to respond to queries confirms control of the address.
+The client SHOULD repeat the discovery process at least every 5 minutes (e.g. by
+limiting the SVCB response TTL) to confirm that the result is still valid, and
+MUST stop using any Encrypted Resolver that is no longer designated.
```

Comment:
What if there is a network outage where packets are dropped? I'm concerned that there could be a transient state, unlucky or induced by an attacker flooding the network, where the client ends up being downgraded after previously having been secure.

Reply:
This is a general concern in either mode.  I've clarified that the discovery process blocks other queries.  Note that this makes the draft sharply incompatible with (hypothetical, horribly broken) SVCB blackhole resolvers.  I'm fine with that.  (Implementers can add an escape hatch if it proves necessary.)

I also added an exemption under "Optimizations" for the case where the client happens to recognize and trust the discovered Encrypted Resolver.